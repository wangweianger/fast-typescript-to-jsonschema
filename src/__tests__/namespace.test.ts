import * as path from 'path';

import genTypeSchema from '../index';

const file = path.join(__dirname, `../../jest/namespace.ts`);
genTypeSchema.genJsonDataFormFile(file);

const getSchema = (type: string) => {
  const schema = genTypeSchema.getJsonSchema(file, type);
  return schema;
};

test('namespace案例_1', () => {
  expect(getSchema('Namespace_1')).toMatchSnapshot();
});

test('namespace案例_2', () => {
  expect(getSchema('Namespace_2')).toMatchSnapshot();
});

test('namespace案例_3', () => {
  expect(getSchema('Namespace_3')).toMatchSnapshot();
});

test('namespace案例_4', () => {
  expect(getSchema('Namespace_4')).toMatchSnapshot();
});

test('namespace案例_5', () => {
  expect(getSchema('Namespace_5')).toMatchSnapshot();
});
