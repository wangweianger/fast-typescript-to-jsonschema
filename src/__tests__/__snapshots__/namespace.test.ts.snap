// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`namespace案例_1 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Param.C.D.E.F.GetBaseDetailResponse": Object {
      "additionalProperties": false,
      "properties": Object {
        "id": Object {
          "type": "string",
        },
      },
      "required": Array [
        "id",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "other1": Object {
      "$ref": "#/definitions/Param.C.D.E.F.GetBaseDetailResponse",
    },
  },
  "required": Array [
    "other1",
  ],
  "type": "object",
}
`;

exports[`namespace案例_2 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Param_1.A": Object {
      "additionalProperties": false,
      "properties": Object {
        "name": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "other1": Object {
      "$ref": "#/definitions/Param_1.A",
    },
  },
  "required": Array [
    "other1",
  ],
  "type": "object",
}
`;

exports[`namespace案例_3 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Param_1.Label": Object {
      "enum": Array [
        1,
        2,
        3,
      ],
      "type": "number",
    },
  },
  "properties": Object {
    "other1": Object {
      "$ref": "#/definitions/Param_1.Label",
    },
  },
  "required": Array [
    "other1",
  ],
  "type": "object",
}
`;

exports[`namespace案例_4 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "Param.C.D.E.F.GetBaseDetailResponse": Object {
      "additionalProperties": false,
      "properties": Object {
        "id": Object {
          "type": "string",
        },
      },
      "required": Array [
        "id",
      ],
      "type": "object",
    },
    "Param_1.A": Object {
      "additionalProperties": false,
      "properties": Object {
        "name": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
    "Param_1.Label": Object {
      "enum": Array [
        1,
        2,
        3,
      ],
      "type": "number",
    },
  },
  "properties": Object {
    "other1": Object {
      "$ref": "#/definitions/Param.C.D.E.F.GetBaseDetailResponse",
    },
    "other2": Object {
      "$ref": "#/definitions/Param_1.A",
    },
    "other3": Object {
      "$ref": "#/definitions/Param_1.Label",
    },
  },
  "required": Array [
    "other1",
    "other2",
    "other3",
  ],
  "type": "object",
}
`;

exports[`namespace案例_5 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "ParamsRule.Level": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "type": "boolean",
        },
        "b": Object {
          "type": "boolean",
        },
        "c": Object {
          "type": "boolean",
        },
        "d": Object {
          "type": "boolean",
        },
      },
      "type": "object",
    },
    "ParamsRule.PAll": Object {
      "additionalProperties": false,
      "properties": Object {
        "level": Object {
          "$ref": "#/definitions/ParamsRule.Level",
        },
        "pObject": Object {
          "$ref": "#/definitions/ParamsRule.PObject",
        },
      },
      "type": "object",
    },
    "ParamsRule.PEnum": Object {
      "enum": Array [
        "enum0",
      ],
      "type": "string",
    },
    "ParamsRule.PObject": Object {
      "additionalProperties": false,
      "properties": Object {
        "a": Object {
          "$ref": "#/definitions/ParamsRule.Level",
        },
        "b": Object {
          "$ref": "#/definitions/ParamsRule.PEnum",
        },
        "c": Object {
          "items": Object {
            "$ref": "#/definitions/ParamsRule.PEnum",
          },
          "type": "array",
        },
      },
      "required": Array [
        "a",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "other1": Object {
      "$ref": "#/definitions/ParamsRule.PAll",
    },
  },
  "required": Array [
    "other1",
  ],
  "type": "object",
}
`;
